hack/ci/e2e-k8s.sh

```diff
@@ -71,12 +71,12 @@ build_with_bazel() {
   bazel shutdown || true
   pkill ^bazel || true
 
-  # ensure the e2e script will find our binaries ...
-  # https://github.com/kubernetes/kubernetes/issues/68306
-  # TODO: remove this, it was fixed in 1.13+
-  mkdir -p '_output/bin/'
-  cp 'bazel-bin/test/e2e/e2e.test' '_output/bin/'
-  PATH="$(dirname "$(find "${PWD}/bazel-bin/" -name kubectl -type f)"):${PATH}"
```

Comment:
why isn't this sufficient? most of the `kube::util::find-binary` logic is for the makefile output.
is there a non-executable file in bazel-bin named `kubectl` now?

Reply:
I suspect it would work fine to use find right now but I'm not very familiar with how we use the bazel cache and future changes (e.g. cross building in the same CI pool) might subtly cause issues. `kube::util::find-binary` is more complex but it does a reasonable job. I don't feel too strongly either way. Yet another option is to use cluster/kubectl.sh like other CI jobs.

Reply:
Pushed a fourth option that doesn't depend on kube::util::find-binary